scripts/proteinInteractionEBI/parseEBI.py

```diff
@@ -0,0 +1,333 @@
+    schemaPieceList.append(curLine)
+    
+    curLine = "name: \"" + dcid + "\""
+    schemaPieceList.append(curLine)
+
+    '''
+    termDic:
+    id:  ['MI:0001']
+    name:  ['interaction detection method']
+    def:  ['Method to determine the interaction']
+    subset:  ['Drugable', 'PSI-MI_slim']
+    synonym:  ['"interaction detect" EXACT PSI-MI-short []']
+    relationship:  ['part_of MI:0000 ! molecular interaction']
+    references:  ['PMID:14755292']
+    parentClassName:  ['MolecularInteraction']
+    '''
+    
+    for key in keyList:
+            
+        if key=="def" and len(termDic[key])>0 :
+            curLine = "description: \"" + termDic[key][0][0].upper() + termDic[key][0][1:] +"\""
+            schemaPieceList.append(curLine)
+
+        elif key=="references" and len(termDic[key])>0:
+            itemList = []
+            for i in range(len(termDic[key])):
+                if termDic[key][i]!="": 
+                    curLine, newReferenceMap = get_references(termDic[key][i])
+                    if curLine:
+                        schemaPieceList.append(curLine)
+                    if newReferenceMap:
+                        newSourceMap[key] = newSourceMap[key].update(newReferenceMap)
+            
+        elif key=="id" and len(termDic[key])>0:       
+            curLine = "psimiID: \"" + termDic[key][0] + "\""
+            schemaPieceList.append(curLine)
+            
+        elif key=="parentClassName" and len(termDic[key])>0:
+            itemList = []
+            for i in range(len(termDic[key])):
+                itemList.append( "dcs:" + termDic[key][i])
+            curLine = "specializationOf: " +  ",".join(itemList)
+            schemaPieceList.append(curLine)
+            
+    curLine = "descriptionUrl: \"http://psidev.info/groups/controlled-vocabularies\""
+    schemaPieceList.append(curLine)
+    
+    return "\n".join(schemaPieceList), termDic['id'][0], dcid, newSourceMap
+
+
+def main(argv):
+    dbFile = argv[0]
+    with open(dbFile, 'r') as fp:
+        file = fp.read()
+    # clip exists in dcs already. Substitute with ClipInteraction
+    file = file.replace("name: clip\ndef", "name: clip interaction\ndef")
+    fileTerms = file.split('\n\n')[1:]
+
+    '''
+    Parsing Steps:
+    1. build the tree by the psi-mi number. A dictionary {psi-mi: node} is used 
+    to access nodes as well.
+    2. save all the tree nodes in the subtree of the three nodes into three set:
+        id: MI:0001 name: interaction detection method
+        id: MI:0190 name: interaction type
+        id: MI:0444 name: database citation
+    3. save the nodes in the three sets to the corresponding enumearation schema
+    '''
+    
+    id2node = {}
+    id2className = {}
+    # build nodes and create the id2node dictionary at first iteration
+    for termText in fileTerms:
+        if not termText.startswith("[Term]"):
+            continue
```

Comment:
Since in all the loops you only care about termText that begins with [Term], how about doing it once after splitting it in line 244.

  fileTerms = file.split('\n\n')[1:]
  fileTerms = [for t in fileTerms if termText.startswith("[Term]")]